app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -138,21 +165,21 @@ def save_transaction_reason_other(self, user_input):
         pass
 
     def is_authorized_pin(self, user_input):
-        pass
+        return self.authorize_pin(user_input)
 
     def is_blocked_pin(self, user_input):
-        pass
+        return self.user.failed_pin_attempts == 3
 
     def is_valid_new_pin(self, user_input):
-        pass
+        return self.is_valid_pin(user_input) and not self.user.check_salt_hashed_password(user_input, self.user.pin)
```

Comment:
just use `self.user.verify_pin(user_input)` instead and make `check_salt_hashed_password` a private method?